server/src/main/kotlin/suwayomi/tachidesk/manga/impl/download/DownloadManager.kt

```diff
@@ -119,8 +126,8 @@ object DownloadManager {
 
     private val notifyFlow = MutableSharedFlow<Unit>(extraBufferCapacity = 1, onBufferOverflow = BufferOverflow.DROP_OLDEST)
 
-    private val statusFlow = MutableSharedFlow<DownloadStatus>()
-    val status = statusFlow.onStart { emit(getStatus()) }
+    private val updatesFlow = MutableSharedFlow<DownloadUpdates>()
+    val updates = updatesFlow.onStart { emit(getDownloadUpdates()) }
```

Comment:
Shouldn't this be getStatus so it sends all current downloads the first time, so the client can cache and use them?
```suggestion
    val updates = updatesFlow.onStart { emit(getStatus()) }
```

Reply:
the initial subscription is one of the main reasons for this change, it causes the n+1 problem, because the subscription does not support data loader batching

to get the full status the download status query should be used and then after that, the subscription informs about changes

Reply:
relying on two separate requests to link up perfectly seems like a bad idea

Reply:
I don't see how that is bad, the subscription is there to inform about changes and not to push the whole status everytime the progress of a single download has changed

you could even ignore the whole download queue and just use the updates from the subscription to show what is currently downloading

Reply:
if the query responds with ```
[
    {id:1234, status:"running"},
    {id:1235, status:"idle"}
]```
but by the time the server deals with the incoming web-socket if 1234 has finished i'tl forever be stuck in the queue on the client

inversely if you do the web-socket first you don't know if the data from the web-socket or the query is more up to date if there is a difference

Reply:
and if we are talking about only sending pieces when something changes then it would make sense to send the whole queue at first
since it is functionally changing from [] (an empty array) to the entire queue on the client
so the changes are the entire queue

Reply:
> not to push the whole status everytime the progress of a single download has changed

i don't see why you are saying this? nobody as said we want this, we already have this currently.

Reply:
> if the query responds with  [ {id:1234, status:"running"}, {id:1235, status:"idle"} ]
but by the time the server deals with the incoming web-socket if 1234 has finished i'tl forever be stuck in the queue on the client
inversely if you do the web-socket first you don't know if the data from the web-socket or the query is more up to date if there is a difference

which I'd say is a rare edge case with a relatively low chance of occurring

> and if we are talking about only sending pieces when something changes then it would make sense to send the whole queue at first
since it is functionally changing from [] (an empty array) to the entire queue on the client
so the changes are the entire queue

the changes are from the servers pov and not the clients?

> i don't see why you are saying this? nobody as said we want this, we already have this currently.

I don't get what you're saying here

> what syer suggested and what would fix the issue i mentioned is having the one web-socket not a web-socket and a query.

it's literally the main reason why I've made this change, the bigger the queue is the worse the responsiveness of the server gets, to literally making it unusable

> the initial subscription is one of the main reasons for this change, it causes the n+1 problem, because the subscription does not support data loader batching

I rather take some unlikely edge case for using a query + subscription compared to this

> if you really want the initial list to be optional

I don't want it at all, neither required nor optional - but I also don't care if something like this needs to be added